### ISSUE TITLE
API proposal : Remove singleRefine from RefinementList

### ISSUE THREAD

**Issue by Jerskouille** [algolia/instantsearch.js#220]
It seems to me that this parameter is redundant with the menu.

Since the menu is already handling disjunctive + single refine, it is here only for conjunctive faceting.
We have two options:
- If we're considering single value attributes => Then it's already the only refinement displayed
- If we're considering array attributes => Then why limit the user in his filtering abilities ?

**Comment by vvo** [comment]
Hey! It's here to expose a feature of the RefinementList (component) to the widget.

So that if you want to build a list of checkboxes where only one checkbox can be ticked at a time, while still displaying the other options then you use refinementList + singleRefine, then you can.

Like "shipping options" (point relais, collissimo..) it fits in it, while using a menu for shipping options may not be obvious to the user.

What do you think?

**Comment by pixelastic** [comment]
> So that if you want to build a list of checkboxes where only one checkbox can be ticked at a time, while still displaying the other options then you use refinementList + singleRefine, then you can.

But isn't exactly what the menu is doing? Having a list where only one item can be checked at a time ? 

In the end I think we need to remove `refinementList` from the exposed widgets. Instead we should expose widgets for each use-case. Maybe we should keep this discussion in #219?

**Comment by pixelastic** [comment]
Back from the other discussion, here are my current thoughts:

- We keep the `refinementList` as a widget (me suggesting removing it was a misunderstanding on my part)
- We remove the `singleRefine` option form the `refinementList` widget (we keep it in the component). We already have the `menu` widget for that. Maybe this is not the best name, but we can work on finding a better name on another issue (`singleSelection`, `radioList`, ...)

**Comment by vvo** [comment]
> We remove the singleRefine option form the refinementList widget (we keep it in the component). We already have the menu widget for that. Maybe this is not the best name, but we can work on finding a better name on another issue (singleSelection, radioList, ...)

I am up for it but also concerned. If I want to make a refinementList out of radio buttons, I will need the singleRefine option. Of course I can use menu to do so, but really menu is sugar on top of refinementList. So from the docs, I would go for refinementList. That's why I am unsure about removing this option.

It's currently not restricting us to do anything right? So let's keep it and see if we need to remove it later on?

**Comment by pixelastic** [comment]
`menu` should be a widget like any other widget, not sugar. The fact that we are internally reusing the same React components should not make it a second-class widget.

`refinementList` should be used only for multiple choices selections (AND or OR). If you want to do a list of radio element, where you can only select one, you should use `menu` instead.

The current `refinementList` widget code is more complex than it should be because we need to ensure that both `operator` and `singleRefine` are mutually exclusive. This should instead be handled directly inside the `menu` widget (which will not expose the `operator` attribute).

**Comment by pixelastic** [comment]
`menu` should be a widget like any other widget, not sugar. The fact that we are internally reusing the same React components should not make it a second-class widget.

`refinementList` should be used only for multiple choices selections (AND or OR). If you want to do a list of radio element, where you can only select one, you should use `menu` instead.

The current `refinementList` widget code is more complex than it should be because we need to ensure that both `operator` and `singleRefine` are mutually exclusive. This should instead be handled directly inside the `menu` widget (which will not expose the `operator` attribute).

**Comment by vvo** [comment]
> The current refinementList widget code is more complex than it should be because we need to ensure that both operator and singleRefine are mutually exclusive. This should instead be handled directly inside the menu widget (which will not expose the operator attribute).

Ok good points! So let's remove it, but we will need a name for this one. And we failed a lot in finding a good one :D

**Comment by tomo-kn** [comment]
I've recently spent a considerable amount of time trying to implement a single-selection filter using the useRefinementList hook, under the assumption that it was the go-to solution for all types of facet filtering in React InstantSearch. It was only after delving deep into the documentation and community resources that I discovered useMenu is actually the recommended approach for single-selection scenarios.

Given this, I believe it would be highly beneficial for the documentation to explicitly clarify the distinction between useRefinementList for multiple selections and useMenu for single selections. A clear indication of this difference could save developers a significant amount of time and confusion.

I understand the flexibility and power of Algolia's InstantSearch hooks and appreciate the wide range of customization options available. However, a straightforward guide or note in the official documentation highlighting the specific use cases for useRefinementList vs. useMenu would be incredibly helpful for those of us integrating Algolia into our React projects.

**Comment by tomo-kn** [comment]
Hello Algolia Team!

I've recently spent a considerable amount of time trying to implement a single-selection filter using the useRefinementList hook, under the assumption that it was the go-to solution for all types of facet filtering in React InstantSearch. It was only after delving deep into the documentation and community resources that I discovered useMenu is actually the recommended approach for single-selection scenarios.

Given this, I believe it would be highly beneficial for the documentation to explicitly clarify the distinction between useRefinementList for multiple selections and useMenu for single selections. A clear indication of this difference could save developers a significant amount of time and confusion.

I understand the flexibility and power of Algolia's InstantSearch hooks and appreciate the wide range of customization options available. However, a straightforward guide or note in the official documentation highlighting the specific use cases for useRefinementList vs. useMenu would be incredibly helpful for those of us integrating Algolia into our React projects.

Best regards

**Comment by Haroenv** [comment]
@tomo-kn, why don't you use `useMenu` for a single-select instead?

**Comment by tomo-kn** [comment]
@Haroenv 
>why don't you use useMenu for a single-select instead?

Yes, what I was trying to convey is that it's difficult to discern from the official documentation that useMenu should be used for single-select scenarios, while useRefinementList is intended for multi-select scenarios. I believe that the documentation for useMenu should include a link to useRefinementList, and vice versa, to make this distinction clearer.
